lib/fibonacci.rb

```diff
@@ -1,9 +1,26 @@
 # Improved Fibonacci
-
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+#
+# Time Complexity - O(n)
+# Space Complexity - O(1)
 # Hint, you may want a recursive helper method
 # Hint:  Fib(0) = 0, Fib(1) = 1, work up from there
+
 def fibonacci(n)
```

Comment:
Nice, although since you have the recursion it's O(n) for space complexity, but you did save space by not needing an n-length array.